Update Credit Card Logos

porada opened the pull request:
With #36850 and #36591 in mind, this PR refactors the credit card images we used and migrates them to Webpack.

## Changes proposed in this Pull Request

* Updated credit card image assets (both regular and disabled states) with detail, performance, and future color updates in mind.
    * Some images used older logos. Those were updated.
    * The markup in some of the older images was so convoluted, it made color updates unnecessarily tricky. Redrawing them simplified it.
* Updated `CreditCard`’s documentation to showcase all supported credit card types and fixed its selected state.
* Removed `CreditCardNumberInput`’s ability to display a credit card logo inside the field based on the input.
    * The supported credit card types are handled by Stripe in their `<iframe>` that overwrites our form fields. In result, we never used that feature anyway.
* Migrated all credit card-related images to Webpack.

### Before

<img src="https://user-images.githubusercontent.com/2060580/67113031-df1d9a80-f1d8-11e9-92be-a266a4dee9b7.png" width="484" height="596" alt="Before">

### After

<img src="https://user-images.githubusercontent.com/2060580/67113086-fc526900-f1d8-11e9-86c1-d6b887d38f1a.gif" width="484" height="596" alt="After">

## Testing instructions

These updates affect 3 components:

1. `CreditCard`
    * Visit the component’s documentation and make sure it looks like on the screenshot above. The _Add card_ button changes the selected item for easier testing.
1. `PaymentLogo`
    * Visit the component’s documentation and make sure the all credit card logos from the screenshot above are still listed (at their 50% size).
1. `CreditCardNumberInput`
    * In the dashboard, click on your avatar, and go to `Manage Purchases`.
    * Go to `Billing History` and `Add Credit Card`.
    * Enter one of the [test credit card details](https://stripe.com/docs/testing#cards) and perform a successful test transaction.

## Other comments

The remaining payment logos and the rest of images located in `static/images/upgrades` will be migrated to Webpack in a separate PR.

porada committed:
Update `CreditCard` component documentation

porada committed:
Fix selected credit card state

porada committed:
Migrate credit card images to Webpack

porada committed:
Update credit card logos

porada commented:
One more thing:

I was trying to make sure `PaymentBox` isn’t affected by those changes, but I’m not sure how to get to it (and its different states). When I read through the code, it doesn’t seem to use any of those images as it uses a gridicon for all credit card types (as a group).

sirreal commented on client/components/upgrades/credit-card-number-input/index.jsx:
```diff
@@ -17,11 +17,45 @@ import Input from 'my-sites/domains/components/form/input';
+
 	render() {
+		const inputStyle = this.props.disabled
+			? {}
```
Can we pass `undefined` rather than empty style object prop?
```suggestion
			? undefined
```

sirreal reviewed (commented):
What I've seen seems to be working well, however I'm having difficulty following testing step 3.

> - Enter each of the test credit card numbers and make sure that an appropriate logo appears next to it.
> - While entering any of the numbers, click on the Security Code info icon and make sure the credit card icon appears in the popover.

I don't see the logo or where to click by "Security Code."

![Screen Shot 2019-10-24 at 18 21 44](https://user-images.githubusercontent.com/841763/67505511-b2133100-f68b-11e9-9f87-91df00077387.png)

porada commented:
It’s possible that some CC forms have it and some don’t. There should be an info icon right after the closing parenthesis. I’ll track down a proper form and update the instructions.

porada commented:
This is weird, because turns out that the icon is present in all forms. An upgrade form loads disabled by default and gets enabled right away (probably when Stripe tells it to). And when the form controls get enabled, the icon suddenly disappears.

porada committed:
:nail_care:

porada commented:
Rebased and updated `CreditCardNumberInput` so that it would show a generic credit card icon regardless of the input state. And like described above, the icon appears at first, but it disappears quickly right after the form gets enabled. The same thing happens to Security Code’s info icon.

What’s suspicious to me is that the text fields also lose some of their styling. `font-family` is no longer set to match the rest of the UI (the system font), but it’s changed to `sans-serif`. It’s all as if the Input component stripped our abstractions on top of it. I recorded a [screencast documenting the issue](https://cloudup.com/cjyjcIanQDr), but I’m not really sure if this PR is the direct cause of that behavior (it’s all fine locally).

@sirreal, is this something you could help me with? Or ping someone who can.

sirreal commented:
It looks like there may be some Stripe libraries that rewrite the html. I see some iframes being used wrapped in elements like `div.__PrivateStripeElement`.

Based on some commit history, maybe @sirbrillig could help or point you in the right direction 🤞

sirbrillig commented:
👋 The card number, expiry, and CVC fields are actually [Stripe Elements](https://stripe.com/payments/elements) and therefore are iframes on to Stripe servers. This change has helped us in many ways, not the least of which is that we no longer have access to the CC information directly which reduces our liability. The only downside is that we have limited options when styling them. Any icon or image which might appear inside the field is something that Stripe does and we probably cannot control.

We do still have our own versions of the CC fields which are displayed while loading the Stripe fields and also for certain payment methods like EBANX. We've tried to style the fields to be nearly the same as the Stripe fields. You can read about [styling the Stripe fields here](https://stripe.com/docs/stripe-js/elements/payment-request-button#styling-the-element).

I hope that's helpful!

porada commented:
@sirbrillig, thanks for chiming in!

The only logos that `CreditCardNumberInput` can display are those supported by Stripe. This means it probably doesn’t make sense for us to maintain them as a feature. In that case, I’ll remove those styling updates.

porada committed:
Revert some of the `CreditCardNumberInput` changes

porada commented:
@sirreal, I made some changes and updated the testing instructions for `CreditCardNumberInput`. The PR is ready for your review.

I stripped the styles responsible for displaying credit card logos inside `CreditCardNumberInput`, because Stripe handles those payment types anyway (and they relied on images outside of Webpack’s pipeline). I’ll create a separate PR suggesting we remove this component abstraction altogether.

sirreal commented:
Rebased.

sirreal reviewed (approved):
Changes test well in devdocs 👍 
Using calypso.live, I was able to add a Credit Card to my account 👍

porada committed:
:nail_care:

porada commented:
@sirreal Thanks for the review!

porada commented on client/components/credit-card/docs/example.jsx:
```diff
@@ -15,30 +15,70 @@ import Button from 'components/button';
 class CreditCards extends Component {
 	state = {
+		selectedCardIndex: 0,
 		cards: [
```
I moved the credit card examples to a constant outside of the class. Let me know if you’re okay with that solution and I’ll carry on with the merge.
